src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -207,12 +207,17 @@ public class JarFile extends ZipFile {
      */
     public static final String MANIFEST_NAME = META_INF + "MANIFEST.MF";
 
+    /**
+     * The 'JAR index' feature has been removed, but we still need to
+     * process existing files which have this entry.
```

Comment:
"but we still need ..."   Maybe better to say that that JarInputStream and the verification of signed JARs need to skip this attribute.

Reply:
I rephrased this to be more specific about which code still needs this. Agree "we" is not the best choice of subject here.